src/management-system-v2/app/(dashboard)/iam/roles/header-actions.tsx

```diff
@@ -87,81 +51,58 @@ const CreateUserModal: FC<{
     setFormatError({});
   }, [form, modalOpen]);
 
-  const submitData = async (
-    values: Record<keyof ApiRequestBody<'/users', 'post'> | 'confirm_password', string>,
-  ) => {
-    debugger;
+  const submitData = async (values: Record<'name' | 'description' | 'expirationDayJs', 'post'>) => {
+    let expiration;
+    if (typeof values.expirationDayJs === 'object')
+      expiration = (values.expirationDayJs as dayjs.Dayjs).toISOString();
+
     try {
-      await postUser({
+      const newRole = await postRole({
         body: {
-          email: values.email,
-          firstName: values.firstName,
-          lastName: values.lastName,
-          username: values.username,
-          password: values.password,
+          name: values.name,
+          description: values.description,
+          expiration,
         },
       });
-      messageApi.success({ content: 'Account created' });
-      close();
+      messageApi.success({ content: 'Role created' });
+      router.push(`/iam/roles/${newRole.id}`);
     } catch (e) {
-      messageApi.error({ content: 'An error ocurred' });
+      messageApi.error({ content: 'Something went wrong' });
     }
   };
 
   return (
-    <Modal open={modalOpen} onCancel={close} footer={null}>
+    <Modal open={modalOpen} onCancel={close} footer={null} title="Create a new role">
       <Form form={form} layout="vertical" onFinish={submitData}>
-        {modalStructureWithoutPassword.map((formField) => (
-          <Form.Item
-            key={formField.dataKey}
-            label={formField.label}
-            name={formField.dataKey}
-            validateStatus={formField.dataKey in formatError ? 'error' : ''}
-            help={formField.dataKey in formatError ? formatError[formField.dataKey] : ''}
-            hasFeedback
-            required
-          >
-            <Input type={formField.type} />
-          </Form.Item>
-        ))}
+        <Form.Item
+          label="Name"
+          name="name"
+          help={formatError.name}
+          validateStatus={formatError.name && 'error'}
+        >
+          <Input />
+        </Form.Item>
 
         <Form.Item
-          name="password"
-          label="Password"
-          rules={[
-            {
-              required: true,
-              message: 'Please input your password!',
-            },
-          ]}
-          validateStatus={'password' in formatError ? 'error' : ''}
-          help={'password' in formatError ? formatError.password : ''}
-          hasFeedback
+          label="Description"
+          name="description"
+          help={formatError.description}
+          validateStatus={formatError.description && 'error'}
         >
-          <Input.Password />
+          <Input.TextArea placeholder="input placeholder" />
```

Comment:
Maybe a better placeholder? :)

Reply:
Agreed. I fixed it in one of the latest commits